pkg/operator/controllers/gameserver_expectations_test.go

```diff
@@ -0,0 +1,86 @@
+package controllers
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	mpsv1alpha1 "github.com/playfab/thundernetes/pkg/operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("expectations tests", func() {
+	// we are performing the tests by including an extra external expectations struct
+	// since the actual one is encapsulated in the GameServerBuild reconciler
+	It("should add and remove game servers to the expectation maps", func() {
+		buildName := "test-build-expectations"
+		gsName1 := "test-gs-expectations-1"
+		gsName2 := "test-gs-expectations-2"
+		buildID := "d82e2163-5912-46cb-9bea-0b783221bb8b"
+		buildNamespace := "default"
+		// create a client to interact with the cluster
+		client := testNewSimpleK8sClient()
+		// create a new expectations struct
+		e := NewGameServerExpectations(client)
+		// create a new GameServerBuild with a single game server
+		err := testCreateGameServerAndBuild(client, gsName1, buildName, buildID, "", mpsv1alpha1.GameServerStateInitializing)
+		Expect(err).ToNot(HaveOccurred())
+		// we're calling the add method manually since the actual expectations class is encapsulated in the controller
+		e.addGameServerToUnderCreationMap(buildName, gsName1)
+		// make sure GameServerBuild and GameServer are in the underCreation map
+		gsUnderCreation, ok := e.gameServersUnderCreation.Load(buildName)
+		Expect(ok).To(BeTrue())
+		mmCreations := gsUnderCreation.(*MutexMap)
+		_, ok = mmCreations.data[gsName1]
+		Expect(ok).To(BeTrue())
+
+		// create a new GameServer and make sure it's added in the map
+		err = client.Create(context.TODO(), testGenerateGameServer(buildName, buildID, buildNamespace, gsName2))
```

Comment:
Usually context.TODO is used as a placeholder during development, context.Background is also an empty context intended for actual use.

Reply:
refactored everywhere, thanks!